### DISCUSSION TITLE
Can I upgrade between major OS releases?

### DISCUSSION THREAD

**Discussion by Letterus** [elementary/triage#120] [Q&A] [answered]
I hope elementary OS 7 based on Ubuntu 22.04 LTS to arrive soon. Since that upgrade would be much appreciated I'm already wondering if there is an upgrade path planned? Will it be possible to upgrade from elementary 6 to 7? Or only afterwards from 7 to 8? For me this question currently is the the main show stopper to use elementary as a daily driver. So any clarification is appreciated. Thanks in advance!

**Comment by kem-a** [comment]
I really hope that devs will consider moving away from release model to rolling model and eventually moving away from Ubuntu base. Fedora would be a great candidate, since it already supports Pantheon desktop out-of-the-box, but Arch could also be considered.

**Reply by Letterus** [reply]
Thank you for your reply. But no, I disagree. There is a rolling app distribution model using flatpak (which I don't really like, but ok, it's the way things go). But there really needs to be stable OS basis imho. Ubuntu is perfectly fine. Maybe Debian would be as well at some point.
But I was asking about the upgrade path, not about changing the distribution model.

**Comment by danirabbit** [comment, marked as answer]
We're actively working on a smooth upgrade feature between major versions but there are many challenges to making it work reliably. In the meantime, we have an upgrade guide that will help you back up and restore your data when installing a new major version here: https://github.com/elementary/os/wiki/Release-Upgrades

**Reply by Letterus** [reply]
Thank you for your reply! Have you got a project or something regarding that upgrade feature? Would dig into it to see if I can help on anything small.

**Reply by meisenzahl** [reply]
@Letterus thanks for offering your help. We discussed a process to do an upgrade and have some prototypes. Right now we have to solve some issues regarding how we package our Flatpak apps: https://github.com/elementary/os/discussions/651

**Reply by danirabbit** [reply]
There is an OS upgrades project here: https://github.com/orgs/elementary/projects/103
